### PR TITLE
feat: add collective rewards

### PR DESCRIPTION
**Pull request by jurajpiar**
# What

Merges all to date contribution regarding the collective rewards feature from [the bim fork](https://github.com/rsksmart/bim-dao-frontend).

> [!WARNING]  
> Requires careful assesment of the applied conflict resolution to avoid bugs/defects and losing functionality
> Also it's advised to merge this as soon as possible, as every new PR created against the current develop will suffer these conflicts

### CODE REVIEW

File: .env.mainnet
Comment (antomor) [inline .env.mainnet:1]: Why `.env.mainnet` isn't in the main repo?

File: .env.regtest
Comment (antomor) [inline .env.regtest:23]: This needs to be changed

File: .env.testnet
Comment (antomor) [inline .env.testnet:22]: we need to change this

File: .env.testnet.local
Comment (antomor) [inline .env.testnet.local:23]: to be changed

File: .env.testnet.qa
Comment (antomor) [inline .env.testnet.qa:1]: Should we add a comment to specify this is used from CR?

File: .env.mainnet
Comment (jessgusclark) [reply .env.mainnet:1]: The file is located here: [.env.prod](https://github.com/RootstockCollective/dao-frontend/blob/develop/.env.prod) and is tied to the CI/CD, let's not change this now.

File: .github/workflows/qa.deploy.yaml
Comment (antomor) [inline .github/workflows/qa.deploy.yaml:1]: I think we shouldn't rename this. Our file can be included with something like `cr.qa.deploy.yaml` or any other name

File: apis.conf
Comment (antomor) [inline apis.conf:1]: It looks like this file is missing in the main repo. Should we remove it also?

File: .env.mainnet
Comment (jurajpiar) [reply .env.mainnet:1]: they use .env.prod. something we need to discuss with @RootstockCollective/dao, really.

File: .env.mainnet
Comment (jessgusclark) [inline .env.mainnet:22]: 404: Not Found

File: .env.mainnet
Comment (jurajpiar) [reply .env.mainnet:1]: imho we should use the way we do it, but there may be a strong argument to keep the dao naming. either way, we need to discuss this together

File: .env.regtest
Comment (jurajpiar) [reply .env.regtest:23]: is this a public repo now?

File: src/components/LoadingSpinner/LoadingSpinner.tsx
Comment (jessgusclark) [inline src/components/LoadingSpinner/LoadingSpinner.tsx:5]: Why did you changes these from export const to export default? This creates extra debugging work and possible confusion in the future when someone does something like this:

```tsx
import whateverNameIWant from './folder/'
```

Where as the way itwas, explicitly makes the developer import in the name of the method:
```tsx
import { LoadingSpinner } from './folder/Filename.txs'
```

File: data.mainnet.json
Comment (jessgusclark) [inline data.mainnet.json:1]: I don't understand the difference between these files and the reference in the .env files. It seems like it should be eitheror.

**Review by jessgusclark — CHANGES_REQUESTED** [review]
I ran though it quickly and left a few comments and one security issue.

File: src/pages/communities/nft/[address].tsx
Comment (antomor) [inline src/pages/communities/nft/[address].tsx:1]: I think we should revert this change

File: src/shared/hooks/useNFTMeta.ts
Comment (antomor) [inline src/shared/hooks/useNFTMeta.ts:1]: Are we sure this file is still used?

File: .env.regtest
Comment (antomor) [reply .env.regtest:23]: Yes, I'd just create a separate PR for the data, so we can update the value here

File: data.mainnet.json
Comment (antomor) [reply data.mainnet.json:1]: That is temporary solution we had to introduce for the MVP. We want to have the possibility to configure it per environment, hence we had to include one file per each environment. The files (so as the env variables) will be removed when we move to v2.

File: package.json
Comment (franciscotobar) [reply package.json:27]: What did you use? Or it wasn't replaced yet?

File: package.json
Comment (franciscotobar) [reply package.json:32]: Its use [here](https://github.com/RootstockCollective/dao-frontend/blob/feature/collective_rewards/next.config.mjs)

Comment (jurajpiar) [reply package.json:32]: yea next.js already has it but as francisco pointed out, in order to use it in the next.config.mjs we need to declare it as a dependency

File: src/app/proposals/hooks/useFetchLatestProposals.ts
Comment (jurajpiar) [reply src/app/proposals/hooks/useFetchLatestProposals.ts:14]: ok, thought it doesn't make sense to have it any shorter than it takes to mine a new block does it.

File: src/components/LoadingSpinner/LoadingSpinner.tsx
Comment (jurajpiar) [reply src/components/LoadingSpinner/LoadingSpinner.tsx:5]: So yea, this is something that needs discussing. Without it we were getting this error.
d8b1dbe6-d847-40c6-9a3d-af74d787b144
To resolve it we make it load in the browser with (nextdynamic)[with https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#nextdynamic], which expects a default export of a component. This solution was found [here](https://stackoverflow.com/a/77824774).

File: src/pages/communities/nft/[address].tsx
Comment (jurajpiar) [reply src/pages/communities/nft/[address].tsx:1]: crap. you're right. I was sure I pressed "accept current", but I guess not.

Comment (jurajpiar) [reply src/pages/communities/nft/[address].tsx:1]: looks like it isn't, tx

File: src/shared/hooks/useNFTMeta.ts
Comment (jurajpiar) [reply src/shared/hooks/useNFTMeta.ts:1]: looks like it isn't, tx

File: package.json
Comment (jessgusclark) [reply package.json]: We were doing checksums and used the methods availabile in ethers.js and viem.

File: apis.conf
Comment (antomor) [reply apis.conf:1]: it's used by CR

File: .env.mainnet
Comment (jessgusclark) [reply .env.mainnet:1]: Given how little time we have, we should aim to make a little changes to existing infrastructure layer as possible. We can always circle back and clean things up like this.

File: .env.mainnet
Comment (jessgusclark) [reply .env.mainnet]: Then for sure this needs to be removed.

File: src/components/LoadingSpinner/LoadingSpinner.tsx
Comment (jessgusclark) [reply src/components/LoadingSpinner/LoadingSpinner.tsx:5]: Okay, that works for me for now.

File: package.json
Comment (jessgusclark) [reply package.json:31]: Great, thanks.

File: src/app/proposals/hooks/useFetchLatestProposals.ts
Comment (jessgusclark) [reply src/app/proposals/hooks/useFetchLatestProposals.ts:14]: I will encourage you to make that argument to them.

File: data.mainnet.json
Comment (jessgusclark) [reply data.mainnet.json:1]: Okay, so the files are placed in the repo, and then referenced via the githibstatic url?

Comment (jurajpiar) [reply data.mainnet.json:1]: yeap .... it's really there just for one month so no point doing anything more... well... reasonable

File: src/app/proposals/hooks/useFetchLatestProposals.ts
Comment (jurajpiar) [reply src/app/proposals/hooks/useFetchLatestProposals.ts:14]: and I shell. I don't see what possible sensible argument they might have for it tbh.

File: .env.mainnet
Comment (jurajpiar) [reply .env.mainnet]: yea just done that. I miscommunicated something. I gave you the link as soon as I resolved conflicts, before making any of the necessary changes. I'm sorry for wasting your time like that. I should have made that clear.

File: .env.testnet.qa
Comment (jurajpiar) [reply .env.testnet.qa:1]: can do.

Comment (jurajpiar) [reply .env.testnet.qa:1]: done

File: .env.mainnet
Comment (jurajpiar) [reply .env.mainnet:1]: ofc, let's not change that now :)

File: .env.testnet
Comment (antomor) [inline .env.testnet:18]:
```suggestion
NEXT_PUBLIC_SIMPLIFIED_REWARD_DISTRIBUTOR_ADDRESS=0x4447EC87854651ECdce3E91e581c34641C91eC46
```

See https://github.com/rsksmart/bim-dao-frontend/pull/71/files

File: .env.testnet.staging
Comment (antomor) [inline .env.testnet.staging:20]:
```suggestion
NEXT_PUBLIC_SIMPLIFIED_REWARD_DISTRIBUTOR_ADDRESS=0x4447EC87854651ECdce3E91e581c34641C91eC46
```

See https://github.com/rsksmart/bim-dao-frontend/pull/71/files

File: public/images/Early-Adopters-Collection-Cover.png
Comment (antomor) [inline public/images/Early-Adopters-Collection-Cover.png:1]: I don't think we need this image

File: src/app/login/Login.tsx
Comment (franciscotobar) [inline src/app/login/Login.tsx:67]: We don't use this, should I remove it?

File: .env.regtest
Comment (Freshenext) [inline .env.regtest:1]: Should this file exist within the repository?
Because, as far as I understand, if this is regtest, I guess the addresses can change depending on the deployment.

(just a thought tho)

File: src/app/user/Balances/hooks/useGetAddressTokens.ts
Comment (Freshenext) [inline src/app/user/Balances/hooks/useGetAddressTokens.ts:29]: Why was this changed to rbtc from RBTC?

File: src/components/Address/AddressOrAlias.tsx
Comment (Freshenext) [inline src/components/Address/AddressOrAlias.tsx:10]: Can this be moved to a separate file?

File: src/components/LoadingSpinner/LoadingSpinner.tsx
Comment (Freshenext) [reply src/components/LoadingSpinner/LoadingSpinner.tsx:5]: Did you try to 'use client' in this component?

**Review by Freshenext — CHANGES_REQUESTED** [review]
Some things to consider, after getting these answers, a lot of testing has to go through in order to approve and merge. 

We might also need documentation on how to get the BE/Contracts running for BIM to test locally

File: .env.regtest
Comment (jurajpiar) [reply .env.regtest:1]: 1. After the release we'll have to have a discussion together about envs.
2. The addresses don't have to necessarily change if the contracts are deployed with CREATE2.
3. We use this for local dev and as such it is much simpler to edit a file than set up env vars manually in each terminal instance every time you want to do something with it.

File: .github/workflows/cr.staging.deploy.yaml
Comment (jurajpiar) [reply .github/workflows/cr.staging.deploy.yaml:1]: my mistake. I thought I returned the original file. will do now.

File: next.config.mjs
Comment (jurajpiar) [reply next.config.mjs:39]: there are two uses:
1. regtest mocks
2. bypass cors check for local dev against testnet

File: src/app/proposals/create/RemoveBuilderProposalForm.tsx
Comment (jurajpiar) [reply src/app/proposals/create/RemoveBuilderProposalForm.tsx:49]: I'm not sure how this made its way back. I'll wrap it in useEffect now.

File: src/app/user/Balances/hooks/useGetAddressTokens.ts
Comment (jurajpiar) [reply src/app/user/Balances/hooks/useGetAddressTokens.ts:29]: Naming convention, I assume. All-caps should be reserved to constants.

File: src/components/Address/AddressOrAlias.tsx
Comment (jurajpiar) [reply src/components/Address/AddressOrAlias.tsx:10]: Can do, but I wouldn't get all worked up about these sorts of things at this point, as the whole codebase is far from pretty, and we want to release asap. Then we'll have time to refactor etc.

File: src/app/login/Login.tsx
Comment (jurajpiar) [reply src/app/login/Login.tsx]: done

File: .github/workflows/cr.staging.deploy.yaml
Comment (jurajpiar) [reply .github/workflows/cr.staging.deploy.yaml:1]: done

File: src/app/proposals/create/RemoveBuilderProposalForm.tsx
Comment (jurajpiar) [reply src/app/proposals/create/RemoveBuilderProposalForm.tsx]: done

**Comment by antomor** [comment]
@Freshenext Thank you for your review 🙏.
I agree with the need of running tests. Let's try to merge this ASAP so we can ask QA to run some regression tests.

>We might also need documentation on how to get the BE/Contracts running for BIM to test locally

As for the instructions to run it locally:
1. We don't have any backend (we use the one already required by the dApp)
2. We tried to document all the steps required the dApp readme
3. For the contracts, in the repo you've the instructions on how to deploy them (even if you shouldn't need to deploy the contracts, if you're running the dApp against Tesnet)
4. If while trying you've some doubts or questions, please feel free to reach out to us and we'll clarify them

File: .env.mainnet
Comment (jessgusclark) [reply .env.mainnet]: I still don't understand here. We can't merge this into the repo if the source of it is private. Is the plan to fix this after we merge this in?

File: README.md
Comment (jessgusclark) [inline README.md:47]: To avoid a cors issue you should be running the server locally. I'm not going to fight it now, but this workaround creates bloat.

File: README.md
Comment (jessgusclark) [inline README.md:102]: dup.

File: README.md
Comment (antomor) [reply README.md:47]: > I'm not going to fight it now
Thank you, let's focus on the deployment now.
I'm creating a document to keep track of all the things we need to agree on

**Comment by jurajpiar** [comment]
Please, @RootstockCollective/dao, let's not focus on details at this stage. Instead, please, ensure that this merge is not jeopardising any functionality that you put in. I'd suggest reviewing only the changes affecting your code and then manually testing that everything works the way you expect it to, as you have more context of your features. On our side, we've done the same.
We can discuss everything else after the release

File: .env.mainnet
Comment (jurajpiar) [reply .env.mainnet]: it will be found once it's merged to develop ;)

**Review by jessgusclark — APPROVED** [review]
I'm good for now knowing that we will clean up the references to the private repo prior to launch, and then work in the following weeks to create some consistency in the code.

File: .env.testnet
Comment (rodrigoncalves) [inline .env.testnet:18]: These new variables wasn't added to all .env files which makes them useless.

Comment (antomor) [reply .env.testnet:18]: @rodrigoncalves good catch 👍🏼 
@jurajpiar it looks like we didn't update the files `.env.dev`, `.env.prod`, `.env.qa`.

Comment (jurajpiar) [reply .env.testnet:18]: done, though I thought we weren't going to touch the orig dao configs.

Comment (rodrigoncalves) [reply .env.testnet:18]: It's still missing some variables in orig dao configs, I can't run the code without them. Are they being replaced by the new ones? The thing is that it's still not clear to me how we are going to do for now on, specially for running locally.

**Review by Freshenext — APPROVED** [review]
Approved due to the priority that this PR has.

File: .github/workflows/cr.qa.deploy.yaml
Comment (lucasvuotto) [inline .github/workflows/cr.qa.deploy.yaml:1]:
```suggestion
name: Deploy to Amazon ECS (CR QA)
```

File: .github/workflows/cr.qa.deploy.yaml
Comment (lucasvuotto) [inline .github/workflows/cr.qa.deploy.yaml:16]: This is not going to fly. https://github.com/RootstockCollective/dao-frontend/blob/develop/Dockerfile#L27 selects the right `.env.*` file for the environment, but it isn't handling `testnet.qa`.

File: .github/workflows/cr.staging.deploy.yaml
Comment (lucasvuotto) [inline .github/workflows/cr.staging.deploy.yaml:1]: Same comments as in `cr.qa.depliy.yaml`.

**Review by lucasvuotto — COMMENTED** [review]
`.env.{dev,prod,qa,testnet}` only include the new variables and they respective workflows are not affected, so, @jessgusclark , this won't interfere with the `dao-frontend` CD.

@jurajpiar left 2 comments for the new workflows, which aren't going to work as they currently stand.

File: .github/workflows/cr.qa.deploy.yaml
Comment (jurajpiar) [reply .github/workflows/cr.qa.deploy.yaml:16]: done

File: .github/workflows/cr.qa.deploy.yaml
Comment (jurajpiar) [reply .github/workflows/cr.qa.deploy.yaml]: done

File: .github/workflows/cr.staging.deploy.yaml
Comment (jurajpiar) [reply .github/workflows/cr.staging.deploy.yaml]: done

**Review by lucasvuotto — APPROVED** [review]
CD bits LGTM

**Pull request merged**
